### ISSUE TITLE
Allow user defineable middlewares to be piped before request decoding

### ISSUE THREAD

**Issue by agustingomes** [chimeraphp/di-symfony#34]
Hi Luis,

I recently opened a PR to solve a problem we're facing: https://github.com/chimeraphp/di-symfony/pull/33 and I wanted to provide you some context on the motivation for it.

Take this request as an example:
```
curl --location --request POST 'localhost:8080/forms' \
--header 'Content-Type: application/json' \
--data-raw '{"invalid"}'
```

Out application currently has an error handler middleware that is added via the following tag: `<tag name="chimera.http_middleware" priority="101"/>`

When the application receives the request, the `BodyParamsMiddleware` tries to decode the request body, and it fails because the input is invalid. 

We would expect the error handler middleware to catch this error, and perform any transformation/reporting needed, however, changing the priority does not help because the middlewares tagged with the `chimera.http_middleware` only get added after the `BodyParamsMiddleware`.

moving the taggable middlewares immediately after the content negotiation middleware, would allow the application to define the necessary middlewares to handle this scenario, and some others as metrics collection.

Let me know if you need further information from my part.

**Comment by rdohms** [comment]
Some important details:
JsonStrategy throws a MalformedBody exception, but that one remains uncaught by anything in Chimera, this leading to a Fatal error.

**Comment by lcobucci** [comment]
I've just realised that this is a duplicate of #4 (:scream:). Let's close it for now and keep things on there since #38 simplifies things quite a lot.

**Comment by lcobucci** [comment]
@rdohms @agustingomes I've just implemented a very simple way to achieve that for v0.4: #86
